Detectors/GlobalTrackingWorkflow/tpcinterpolationworkflow/include/TPCInterpolationWorkflow/TPCResidualAggregatorSpec.h

```diff
@@ -178,11 +181,22 @@ class ResidualAggregatorDevice : public o2::framework::Task
   void updateTimeDependentParams(ProcessingContext& pc)
   {
     o2::base::GRPGeomHelper::instance().checkUpdates(pc);
+    mTPCVDriftHelper.extractCCDBInputs(pc);
     if (!mInitDone) {
-      mInitDone = true;
       mAggregator->setDataTakingContext(pc.services().get<DataTakingContext>());
+      mAggregator->setOrbitResetTime(o2::base::GRPGeomHelper::instance().getOrbitResetTimeMS());
+      if (mTPCVDriftHelper.isUpdated()) {
+        LOGP(info, "Updating TPC fast transform map with new VDrift factor of {} wrt reference {} and DriftTimeOffset correction {} wrt {} from source {}",
+             mTPCVDriftHelper.getVDriftObject().corrFact, mTPCVDriftHelper.getVDriftObject().refVDrift,
+             mTPCVDriftHelper.getVDriftObject().timeOffsetCorr, mTPCVDriftHelper.getVDriftObject().refTimeOffset,
+             mTPCVDriftHelper.getSourceName());
+        mAggregator->setTPCVDrift(mTPCVDriftHelper.getVDriftObject());
+        mTPCVDriftHelper.acknowledgeUpdate();
+      }
+      mInitDone = true;
```

Comment:
Should this only be set if the vdrift object is actually propagated to the aggregator? Otherwise there might be a risk it is not set, or?

Reply:
I am not sure if it can be that it's actually not set? In my local test this worked, but that does not mean it always does of course... And I am still trying to get it less verbose, currently there seem to be quite some log messages from the TPC vDrift helper and I actually need to query the reference vDrift only once